### PR TITLE
Fix Observation Fileds

### PR DESCRIPTION
**Pull request by MonishaSivanathan**
- fixed units field
- added the missing fields like interpretation, reference range

### CODE REVIEW

File: integration-pipeline/HL7v2/resources/observation.py [hunk 1]
```diff
@@ -162,6 +163,16 @@ def prepare_observation(
         observation.performer.append(Reference(reference="Organization/" + organization.id))
 
         organizations.append(organization)
+    
+    if "referenceRange" in data and "range" in data["referenceRange"]:
+        observation.referenceRange = [Observation_ReferenceRange(
+            text=data["referenceRange"]["range"]
+        )]
+
+    if "interpretation" in data and "flag" in data["interpretation"] and len(data["interpretation"]["flag"]) > 0:
```
Comment (tamsud) [inline integration-pipeline/HL7v2/resources/observation.py:172]: support multiple index as well

File: integration-pipeline/HL7v2/resources/observation.py [hunk 2]
```diff
@@ -118,9 +118,10 @@ def prepare_observation(
     if "string" in data.get("value", {}):
         observation.valueString = " ".join(data["value"]["string"])
 
-    if "unit" in data.get("value", {}):
+    if "units" in data.get("value", {}) and "number" in data.get("value", {}):
         observation.valueQuantity = Quantity(
-            value=data["value"]["TX"], unit=data["value"]["code"]
+            value=float(data["value"]["number"][0]),
```
Comment (tamsud) [inline integration-pipeline/HL7v2/resources/observation.py:123]: check the index if its valid then try to access the index position

File: integration-pipeline/HL7v2/resources/observation.py [hunk 3]
```diff
@@ -118,9 +118,10 @@ def prepare_observation(
...
+            unit=data["value"]["units"]["code"]
```
Comment (tamsud) [inline integration-pipeline/HL7v2/resources/observation.py:124]: check the key before access it.

**Review by tamsud — CHANGES_REQUESTED** [review]

File: integration-pipeline/HL7v2/resources/observation.py (hunk 1 above)
Comment (MonishaSivanathan) [reply integration-pipeline/HL7v2/resources/observation.py:172]: fixed

File: integration-pipeline/HL7v2/resources/observation.py (hunk 2 above)
Comment (MonishaSivanathan) [reply integration-pipeline/HL7v2/resources/observation.py:123]: changed

**Pull request merged**